File: articles/can-grit-be-taught/go/can_grit_be_taught_effect_size.go
package main

import "fmt"

func CohensD(meanTreatment, meanControl, pooledSD float64) float64 {
	if pooledSD == 0 {
		return 0
	}
	return (meanTreatment - meanControl) / pooledSD
}

func main() {
	d := CohensD(4.2, 3.6, 0.8)
	fmt.Printf("Synthetic intervention effect size estimate: %.3f\n", d)
	fmt.Println("Professional caution: synthetic demonstration only.")
}

File: articles/can-grit-be-taught/fortran/can_grit_be_taught_effect_size.f90
program can_grit_be_taught_effect_size
  implicit none
  real :: mean_treatment, mean_control, pooled_sd, d

  mean_treatment = 4.2
  mean_control = 3.6
  pooled_sd = 0.8

  if (pooled_sd == 0.0) then
    d = 0.0
  else
    d = (mean_treatment - mean_control) / pooled_sd
  end if

  print *, "Synthetic intervention effect size estimate:", d
  print *, "Professional caution: synthetic demonstration only."
end program can_grit_be_taught_effect_size
